src/adc.rs

```diff
@@ -45,6 +45,12 @@ pub struct AdcBuffer {
     buffer: [u8; BUFFER_SIZE],
 }
 
+impl AsMut<[u8]> for AdcBuffer {
```

Comment:
You no longer need this.

Reply:
I would prefer if all `AsRef` and `AsMut` implementations were removed until we see the need for them.